modules/apps/27-interchain-accounts/types/packet.go

```diff
@@ -66,27 +66,32 @@ The Memo format is defined like so:
 {
 	// ... other memo fields we don't care about
 	"callbacks": {
-		"src_callback_address": {contractAddrOnSrcChain},
+		"src_callback_address": {contractAddrOnSourceChain},
 		"dest_callback_address": {contractAddrOnDestChain},
-		"src_callback_msg": {jsonObjectForSrcChainCallback}, // optional field
-		"dest_callback_msg": {jsonObjectForDestChainCallback}, // optional field
-	}
 
+		// optional fields
+		"src_callback_msg": {jsonObjectForSourceChainCallback},
+		"dest_callback_msg": {jsonObjectForDestChainCallback},
+	}
 }
 ```
 
 */
 
-// GetSrcCallbackAddress returns the callback address on the source chain.
-// ADR-8 middleware should callback on the sender address on the source chain
-// if the sender address is an IBC Actor (i.e. smart contract that accepts IBC callbacks)
-func (iapd InterchainAccountPacketData) GetSrcCallbackAddress() string {
+// GetSourceCallbackAddress returns the source callback address provided in the packet data memo.
+// If no callback address is specified, an empty string is returned.
+//
+// The memo is expected to specify the callback address in the following format:
+// { "callbacks": { "src_callback_address": {contractAddrOnSourceChain}}
+//
+// ADR-8 middleware should callback on the returned address if it is a PacketActor
+// (i.e. smart contract that accepts IBC callbacks).
+func (iapd InterchainAccountPacketData) GetSourceCallbackAddress() string {
```

Comment:
What would be returned if the json struct has 2 `callbacks` entries or 2 `src_callback_address` fields inside the same `callbacks` object?

Reply:
The second entry will likely override the first if there is two identical keys in the stringified JSON object.

https://go.dev/play/p/SQQBSfr2gDq